### PR TITLE
#SB-1485 :Org and course dashboard migration

### PR DESCRIPTION
**Pull request by nilesh-more**

### CODE REVIEW

File: src/app/migration/src/package.json [hunk 1]
```diff
@@ -25,9 +25,13 @@
     "@angular/platform-browser-dynamic": "^5.0.0",
     "@angular/router": "^5.2.1",
     "angular2-uuid": "^1.1.1",
+    "bootstrap": "^3.3.7",
```
Comment (HarishGangula) [inline src/app/migration/src/package.json]: Remove Bootstrap

File: src/app/migration/src/src/app/app.module.ts [hunk 2]
```diff
@@ -27,24 +39,37 @@ import { ProfileViewComponent } from './profile/profile-view/profile-view.compon
     MainMenuComponent,
     SearchComponent,
     CommunityListComponent,
+    CourseConsumptionDashboardComponent,
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/app.module.ts]: All components should be at same place

File: src/app/migration/src/src/app/dashboard/course-consumption/course-consumption.component.css [hunk 3]
```diff
@@ -0,0 +1,57 @@
+.margin-bottom-10 {
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/dashboard/course-consumption/course-consumption.component.css]: Remove !important from css

File: src/app/migration/src/src/app/dashboard/course-consumption/course-consumption.component.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	getData(timePeriod: string, identifier: string) {
+		this.showLoader = true
+		this.timePeriod = timePeriod ? timePeriod : '7d'
+		this.identifier = identifier
+		this.CourseConsumptionService.getData({
+			identifier: this.identifier,
+			timePeriod: this.timePeriod
+		}).subscribe(
+			data => {
+				console.log('API-Response: Dashboard -', data)
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/dashboard/course-consumption/course-consumption.component.ts]: No console logs

File: src/app/migration/src/src/app/dashboard/datasource/organisation.service.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import { DataService } from '../../services/data/data.service'
+import { HttpClient } from '@angular/common/http';
+import { DashboardUtilsService } from '../../dashboard/datasource/dashboard-utils.service'
+import * as _ from 'lodash';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/observable/throw';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class OrganisationService extends DataService  {
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/dashboard/datasource/organisation.service.ts]: missing documentation

File: src/app/migration/src/src/app/dashboard/organisation/organisation.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/dashboard/organisation/organisation.component.spec.ts]: Test cases should contain proper assertions

**Review by HarishGangula — CHANGES_REQUESTED** [review]
Update the documentation using https://compodoc.github.io/compodoc/ and all test case should contain proper assertions and test cases should cover the all the aspects of the feature

File: src/app/migration/src/package.json (hunk 1 above)
Comment (nilesh-more) [reply src/app/migration/src/package.json]: resolved

File: src/app/migration/src/src/app/app.module.ts (hunk 2 above)
Comment (nilesh-more) [reply src/app/migration/src/src/app/app.module.ts]: noted...

File: src/app/migration/src/src/app/dashboard/course-consumption/course-consumption.component.css (hunk 3 above)
Comment (nilesh-more) [reply src/app/migration/src/src/app/dashboard/course-consumption/course-consumption.component.css]: resolved

File: src/app/migration/src/src/app/dashboard/course-consumption/course-consumption.component.ts (hunk 4 above)
Comment (nilesh-more) [reply src/app/migration/src/src/app/dashboard/course-consumption/course-consumption.component.ts]: resolved

File: src/app/migration/src/src/app/dashboard/datasource/organisation.service.ts (hunk 5 above)
Comment (nilesh-more) [reply src/app/migration/src/src/app/dashboard/datasource/organisation.service.ts]: @souravdey091, please resolve this

File: src/app/migration/src/src/app/dashboard/organisation/organisation.component.spec.ts (hunk 6 above)
Comment (nilesh-more) [reply src/app/migration/src/src/app/dashboard/organisation/organisation.component.spec.ts]: @souravdey091

**Comment by HarishGangula** [comment]
@nilesh-more @souravdey091  take pull from main branch and update the documentation,
for any help revert me back

File: src/app/migration/src/src/app/modules/dashboard/components/course-consumption/course-consumption.component.html [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+                        <div class="meta dashBoardCardText">{{block.name}}</div>
+                        <h2 class="description dashboardCardSubText">{{block.value}}</h2>
+                    </div>
+                </div>
+            </div>
+            <div class="ui one column grid margin-top-20" *ngIf="graphData && graphData.length > 0">
+                <div class="column">
+                    <div class="regular slider padding-top-30 course-graph-holder">
+                        <div *ngFor="let block of graphData;let graphIndex = index;">
+                            <div [ngStyle]="{'display':showGraph === graphIndex ? '' : 'none' }">
```
Comment (HarishGangula) [inline src/app/migration/src/src/app/modules/dashboard/components/course-consumption/course-consumption.component.html]: Fix this visibility  use hidden dom attribute

**Comment by HarishGangula** [comment]
1. Remove frame*.js files
2. solve linting problems
3. fix breadcrumbs with UI
4. replace href -> routerLink for inner pages of angular app
5. imports should come from module
6. use moment js
7. revisit html templates to optimize

**Review by HarishGangula — CHANGES_REQUESTED** [review]

**Pull request closed without merging**